File: rulepacks/2026.08.3/testdata/security.cs
// Fixture de semgrep --test — no es código del producto.
using System.Runtime.Serialization.Formatters.Binary;
using System.Text.Json;

namespace Fixtures
{
    public class SeguridadFixture
    {
        public object Deserializar()
        {
            // ruleid: csharp-binaryformatter
            var formateador = new BinaryFormatter();
            return formateador;
        }

        public string SerializarSeguro(object dato)
        {
            // ok: csharp-binaryformatter
            return JsonSerializer.Serialize(dato);
        }
    }
}

File: rulepacks/2026.08.3/testdata/security.go
// Fixture de semgrep --test — no es código del producto.
package fixtures

import (
	"database/sql"
	"strconv"
)

func buscarUsuario(db *sql.DB, id int) {
	// ruleid: go-sql-concat
	rows, err := db.Query("SELECT nombre FROM usuarios WHERE id = " + strconv.Itoa(id))
	_ = rows
	_ = err
}

func buscarUsuarioSeguro(db *sql.DB, id int) {
	// ok: go-sql-concat
	rows, err := db.Query("SELECT nombre FROM usuarios WHERE id = ?", id)
	_ = rows
	_ = err
}

func exportarEventos(db *sql.DB, repo string) {
	consulta := "SELECT id FROM eventos WHERE repo = '" + repo + "'"
	// ruleid: go-sql-concat-en-variable
	rows, err := db.Query(consulta)
	_ = rows
	_ = err
}

func listarEventos(db *sql.DB) {
	consulta := "SELECT id FROM eventos"
	consulta += " ORDER BY fecha DESC"
	// ok: go-sql-concat-en-variable
	rows, err := db.Query(consulta)
	_ = rows
	_ = err
}
